src/rpc/mining.cpp

```diff
@@ -124,22 +124,40 @@ UniValue generateBlocks(boost::shared_ptr<CReserveScript> coinbaseScript, int nG
             LOCK(cs_main);
             IncrementExtraNonce(pblock, chainActive.Tip(), nExtraNonce);
         }
-        // Dogecoin: Don't mine Aux blocks in regtest
-        //CAuxPow::initAuxPow(*pblock);
-        //CPureBlockHeader& miningHeader = pblock->auxpow->parentBlock;
-        while (nMaxTries > 0 && pblock->nNonce < nInnerLoopCount && !CheckProofOfWork(pblock->GetPoWHash(), pblock->nBits, Params().GetConsensus(nHeight))) {
-            ++pblock->nNonce;
-            --nMaxTries;
+        if (!nMineAuxPow) {
+            while (nMaxTries > 0 && pblock->nNonce < nInnerLoopCount && !CheckProofOfWork(pblock->GetPoWHash(), pblock->nBits, Params().GetConsensus(nHeight))) {
+                ++pblock->nNonce;
+                --nMaxTries;
+            }
+        } else {
+            CAuxPow::initAuxPow(*pblock);
+            CPureBlockHeader& miningHeader = pblock->auxpow->parentBlock;
+            while (nMaxTries > 0 && miningHeader.nNonce < nInnerLoopCount && !CheckProofOfWork(miningHeader.GetHash(), pblock->nBits, Params().GetConsensus(nHeight))) {
```

Comment:
`miningHeader.GetHash()` should be `miningHeader.GetPoWHash()` and then it works for me on testnet too!

https://chain.so/block/DOGETEST/1ab471a198f54310c98e5820a4766894f1cec53e654cb03e25082b387918b78b

Reply:
Sweet, regtest as well:

```0401620015f0efb3eb3f094999beed2ebbd7253d184100e45894b2a0a345b3f30b8ad89dacda17c77120627d4969798138e7cafc1f2602ee75741e707f88f4b68e8f52f248d8ab60ffff7f200000000001000000010000000000000000000000000000000000000000000000000000000000000000ffffffff292811ff2956195106bbde32cc1ffe044c4b25e199e3d585ffc943b43d3950e22c9c0100000000000000ffffffff0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000010000000000000000000000000000000000000000000000000000000000000000000000c1ce4ca223f2e0f951478af443777be87efed1f4b21b31d3c1217c697086ab8a0000000000000000000000000101000000010000000000000000000000000000000000000000000000000000000000000000ffffffff0401380101ffffffff0200203d88792d00001976a914142b5e7a7f8728324db776a7257200e2d02db1d588ac0000000000000000266a24aa21a9ede2f61c3f71d1defd3fa999dfa36953755c690689799962b48bebd836974e8cf900000000```